python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py

```diff
@@ -207,12 +346,139 @@ def parse_stream_tables(
         return tables
 
 
+class AirbyteInstanceCacheableAssetsDefintion(CacheableAssetsDefinition):
+    def __init__(
+        self,
+        airbyte_resource_def: ResourceDefinition,
+        workspace_id: Optional[str],
+        key_prefix: List[str],
+        create_assets_for_normalization_tables: bool,
+        connection_to_group_fn: Optional[Callable[[str], Optional[str]]],
+    ):
+        self._airbyte_resource_def = airbyte_resource_def
+        self._airbyte_instance: AirbyteResource = airbyte_resource_def(
+            build_init_resource_context()
+        )
+
+        self._workspace_id = workspace_id
+        self._key_prefix = key_prefix
+        self._create_assets_for_normalization_tables = create_assets_for_normalization_tables
+        self._connection_to_group_fn = connection_to_group_fn
+
+        super().__init__(unique_id="airbyte")
+
+    def get_cached_data(self) -> Sequence[CachedAssetsData]:
+
+        workspace_id = self._workspace_id
+        if not workspace_id:
+            workspaces = cast(
+                List[Dict[str, Any]],
+                check.not_none(
+                    self._airbyte_instance.make_request(endpoint="/workspaces/list", data={})
+                ).get("workspaces", []),
+            )
+
+            check.invariant(len(workspaces) <= 1, "Airbyte instance has more than one workspace")
+            check.invariant(len(workspaces) > 0, "Airbyte instance has no workspaces")
+
+            workspace_id = workspaces[0].get("workspaceId")
+
+        connections = cast(
```

Comment:
not a huge deal, but you could probably avoid these casts with `check.inst(..., dict)` (not 100% sure)

Reply:
Looks like mypy isn't satisfied with this. Will leave for now even though it's a little gross.